aws/getOrBuildDockerImage.ts

```diff
@@ -98,6 +102,7 @@ export const getOrBuildDockerImage =
 			debug?.(`Image tag ${imageTag} exists on ${repositoryUri}`)
 			return { imageTag }
 		} else {
+			if (beforeBuild !== undefined) await beforeBuild()
```

Comment:
Instead of inventing your own event messaging, use an EventEmitter. https://nodejs.org/api/events.html#class-eventemitter

Reply:
Event emitter cannot ensure the required file is successfully copied over.